From d4a86565be9f27e9a66e7ff5078c060067e708ee Mon Sep 17 00:00:00 2001
From: Fancy Fang <chen.fang@nxp.com>
Date: Fri, 22 Sep 2017 22:41:53 +0800
Subject: [PATCH 2585/5242] MLK-16536-11 video: fbdev: dcss: add
 '__maybe_unused' to 'dtg_irq_mask()'
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

commit  1457545a043d329d8cd7328d680dec248ca05695 from
https://source.codeaurora.org/external/imx/linux-imx.git

The function 'dtg_irq_mask()' is not used at this
moment which caused gcc compiler generate the build
warning:

"‘dtg_irq_mask’ defined but not used [-Wunused-function]"

So add attribute '__maybe_unused' to 'dtg_irq_mask()'
functon definition to avoid this build warning.

Signed-off-by: Fancy Fang <chen.fang@nxp.com>
Reviewed-by: Robby Cai <robby.cai@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/video/fbdev/mxc/imx_dcss.c |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/video/fbdev/mxc/imx_dcss.c b/drivers/video/fbdev/mxc/imx_dcss.c
index 584bfa9..f36a691 100644
--- a/drivers/video/fbdev/mxc/imx_dcss.c
+++ b/drivers/video/fbdev/mxc/imx_dcss.c
@@ -2181,8 +2181,8 @@ static void ctxld_irq_unmask(uint32_t irq_en, struct dcss_info *info)
 	writel(irq_en, info->base + chans->ctxld_addr + CTXLD_CTRL_STATUS_SET);
 }
 
-static void dtg_irq_mask(unsigned long hwirq,
-			 struct dcss_info *info)
+static void __maybe_unused dtg_irq_mask(unsigned long hwirq,
+					struct dcss_info *info)
 {
 	unsigned long irq_mask = 0;
 	struct dcss_channels *chans = &info->chans;
-- 
1.7.9.5

